### PR TITLE
tests: add dump plugin integration test (CRAFT-70)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Make sure source files are installed and craft-parts work directories
are ignored.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by sergiusens — APPROVED** [review]
Looks good, this just made me think we might want to rename "source-subdir" to "build-workdir" or something less confusing and on point.

File: tests/integration/plugins/test_dump.py [hunk 1]
```diff
@@ -1 +1,84 @@
...
+    lf = craft_parts.LifecycleManager(parts, application_name="test_dump")
+
+    with lf.action_executor() as ctx:
+        ctx.execute(Action("foo", Step.PULL))
+        ctx.execute(Action("foo", Step.BUILD))
+
+    install_dir = Path("parts", "foo", "install")
+
+    # only the file in subdir should be installed
+    assert list(install_dir.rglob("*")) == [install_dir / "foobar.txt"]
```
Comment (facundobatista) [inline tests/integration/plugins/test_dump.py:51]: I didn't know about `rglob`! nice!

**Review by facundobatista — APPROVED** [review]
Nice!

**Comment by cmatsuoka** [comment]
> Looks good, this just made me think we might want to rename "source-subdir" to "build-workdir" or something less confusing and on point.

We have many small things to adjust before 1.0, renaming things will be very easy to do.

**Pull request merged**